packages/runed/src/lib/utilities/useNetworkStatus/useNetworkStatus.svelte.ts

```diff
@@ -0,0 +1,146 @@
+import { addEventListener } from "$lib/internal/utils/event.js";
+import { Previous } from "$lib/utilities/index.js";
+
+/**
+ * @desc The `NetworkInformation` interface of the Network Information API
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation
+ */
+type NetworkInformation = {
+	readonly downlink: number;
+	readonly downlinkMax: number;
+	readonly effectiveType: "slow-2g" | "2g" | "3g" | "4g";
+	readonly rtt: number;
+	readonly saveData: boolean;
+	readonly type:
+		| "bluetooth"
+		| "cellular"
+		| "ethernet"
+		| "none"
+		| "wifi"
+		| "wimax"
+		| "other"
+		| "unknown";
+} & EventTarget;
+
+type NavigatorWithConnection = Navigator & { connection: NetworkInformation };
+
+interface NetworkStatus {
+	/**
+	 * @desc Returns the online status of the browser.
+	 * @see https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine
+	 */
+	online: boolean;
+	/**
+	 * @desc The {Date} object pointing to the moment when state update occurred.
+	 */
+	updatedAt: Date;
+	/**
+	 * @desc Effective bandwidth estimate in megabits per second, rounded to the
+	 * nearest multiple of 25 kilobits per seconds.
+	 * @see https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/downlink
+	 */
+	downlink?: NetworkInformation["downlink"];
+	/**
+	 * @desc Maximum downlink speed, in megabits per second (Mbps), for the
+	 * underlying connection technology
+	 * @see https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/downlinkMax
+	 */
+	downlinkMax?: NetworkInformation["downlinkMax"];
+	/**
+	 * @desc Effective type of the connection meaning one of 'slow-2g', '2g', '3g', or '4g'.
+	 * This value is determined using a combination of recently observed round-trip time
+	 * and downlink values.
+	 * @see https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType
+	 */
+	effectiveType?: NetworkInformation["effectiveType"];
+	/**
+	 * @desc Estimated effective round-trip time of the current connection, rounded
+	 * to the nearest multiple of 25 milliseconds
+	 * @see https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/rtt
+	 */
+	rtt?: NetworkInformation["rtt"];
+	/**
+	 * @desc {true} if the user has set a reduced data usage option on the user agent.
+	 * @see https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/saveData
+	 */
+	saveData?: NetworkInformation["saveData"];
+	/**
+	 * @desc The type of connection a device is using to communicate with the network.
+	 * It will be one of the following values:
+	 *  - bluetooth
+	 *  - cellular
+	 *  - ethernet
+	 *  - none
+	 *  - wifi
+	 *  - wimax
+	 *  - other
+	 *  - unknown
+	 *  @see https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/type
+	 */
+	type?: NetworkInformation["type"];
+}
+
+/**
+ * Tracks the state of browser's network connection.
+ * @returns null if the function is not run in the browser
+ */
+export function useNetworkStatus() {
```

Comment:
```suggestion
export class NetworkStatus {
```

Let's refactor this to be a class. Rather than having the `.current`, we can use the `IsSupported` Runed util to determine if it's a supported environment. We can expose `isSupported` as a reactive property on the class, and then each of the individual parts as well.

This way it could be used like:

```ts
const netStat = new NetworkStatus()

if (netStat.isSupported) {
	console.log(netStat.online)
	// etc...
}
```

Reply:
Thanks a lot! I'm pretty new to Svelte 5 (especially with classes), but I decided to give it a try. 
I've rewritten NetworkStatus to a class, as you suggested, and updated the demo. I haven't updated the docs yet, but I can do that once we settle on the implementation.
Let me know how we can improve it further, I'll be happy to refactor it :)